src/lib/linux-api/src/time.rs

```diff
@@ -118,6 +118,52 @@ pub fn alarm(secs: u32) -> Result<u32, Errno> {
     Ok(res.try_into().unwrap())
 }
 
+/// Make a `getitimer` syscall.
+///
+/// # Safety
+///
+/// `curr_value` must be safe for the kernel to write to.
+pub unsafe fn getitimer_raw(which: i32, curr_value: *mut itimerval) -> Result<(), Errno> {
+    unsafe { syscall!(linux_syscall::SYS_getitimer, which, curr_value) }
+        .check()
+        .map_err(Errno::from)
+}
+
+/// Make a `getitimer` syscall.
+pub fn getitimer(which: ITimerId, curr_value: &mut itimerval) -> Result<(), Errno> {
+    unsafe { getitimer_raw(which.into(), curr_value) }
+}
+
+/// Make a `setitimer` syscall.
+///
+/// # Safety
+///
+/// `old_value` must be safe for the kernel to write to, or NULL.
+///
+/// An invalid or inaccessible `new_value` *isn't* a safety violation, but may
+/// cause the syscall to fail e.g. with `EFAULT`.
+pub unsafe fn setitimer_raw(
+    which: i32,
+    new_value: *const itimerval,
+    old_value: *mut itimerval,
+) -> Result<(), Errno> {
+    unsafe { syscall!(linux_syscall::SYS_setitimer, which, new_value, old_value) }
```

Comment:
In the kernel the syscall takes arguments of type `struct __kernel_old_itimerval`. Is this the same as `linux_itimerval` for x86-64, or should this function take `kernel_old_timeval` instead? (Even if they're the same, I think it would be better to have it take `kernel_old_timeval` instead just to be consistent with the kernel.)

It also takes `which` as an `int`; should we keep it as an `int` here in the wrapper too? I don't have a strong preference here though, since realistically we only target x86-64 where `int == i32`.

Reply:
Done